mozilla/fxa
# feat(react): Implement OAuth functionality for Signin flows

## Because
* We're converting our legacy Backbone pages to React incrementally

## This pull request:
* Adds the 'oauth/signin' route
* Adds React views for 'signin_token_code' and 'signin_totp_code' to content-server router
* Includes some navigation utils shuffling and additions to accommodate the OAuth flow with our OAuth hook in signin flow components
* Tweaks OAuth integration.wantsKeys to return true if sync
* Makes 'keys' params in oauth hook optional, renames unwrapKB arg in oauth hook for consistency
* Removes 'login_hint' check from 123done prompt=none since it's no longer required
* Updates unit tests for Signup, Signin, ConfirmSignupCode, SigninTokenCode, SigninTotpCode
* Removes brand messaging portal from React pages
* Update TODOs with ticket numbers

## Issue that this pull request solves
closes FXA-6518

We will likely need a follow up issue for implementation on inline TOTP / recovery set up for after that lands (assigned to Barry).

## Checklist


## Other information (Optional)

Manual testing suggestions:
- Test regular web flow (no auth) to verify still works as expected
- Test with OAuth (123done) and:
  - with/without cached login
  - with `forceGlobally` set to `true` to force verification with token/totp
  - with/without 2FA enabled for the account
